apps/tlon-mobile/src/screens/Onboarding/SignUpPasswordScreen.tsx

```diff
@@ -178,111 +156,98 @@ export const SignUpPasswordScreen = ({
-            </ListItem.EndContent>
-          </ListItem>
+        <YStack gap="$m" paddingHorizontal="$2xl" paddingVertical="$l">
+          {/* {lureMeta ? <AppInviteDisplay metadata={lureMeta} /> : null} */}
+          <View padding="$xl">
+            <TextV2.Text size="$body" color="$primaryText">
+              Please set a strong password with at least 8 characters.
+            </TextV2.Text>
+          </View>
+          <YStack gap="$2xl">
+            <Controller
+              control={control}
+              name="password"
+              rules={{
+                required: 'Password must be at least 8 characters.',
+                minLength: {
+                  value: 8,
+                  message: 'Password must be at least 8 characters.',
+                },
+              }}
+              render={({ field: { onChange, onBlur, value } }) => (
+                <Field
+                  label="Password"
+                  error={errors.password?.message}
+                  paddingTop="$m"
+                >
+                  <TextInput
+                    placeholder="Choose a password"
+                    onBlur={onBlur}
+                    onChangeText={onChange}
+                    onSubmitEditing={() => setFocus('confirmPassword')}
+                    value={value}
+                    secureTextEntry
+                    autoCapitalize="none"
+                    autoCorrect={false}
+                    returnKeyType="next"
+                    enablesReturnKeyAutomatically
+                  />
+                </Field>
+              )}
+            />
+            <Controller
+              control={control}
+              name="confirmPassword"
+              rules={{
+                required: 'Enter the password again for confirmation.',
+                validate: (value, { password }) =>
+                  value === password || 'Passwords must match.',
+              }}
+              render={({ field: { onChange, onBlur, value, ref } }) => (
+                <Field
+                  paddingTop="$m"
+                  label="Confirm Password"
+                  error={errors.confirmPassword?.message}
+                >
+                  <TextInput
+                    placeholder="Confirm password"
+                    onBlur={onBlur}
+                    onChangeText={onChange}
+                    onSubmitEditing={onSubmit}
+                    value={value}
+                    secureTextEntry
+                    autoCapitalize="none"
+                    autoCorrect={false}
+                    returnKeyType="send"
+                    ref={ref}
+                    enablesReturnKeyAutomatically
+                  />
+                </Field>
+              )}
+            />
+          </YStack>
+          <TextV2.Text padding="$xl" size="$label/s" color="$tertiaryText">
+            By registering you agree to Tlon&rsquo;s{' '}
+            <TextV2.RawText
```

Comment:
This should link to or otherwise open the EULA.

Reply:
Fixed